build(nix): change the pkgs to final, add new version of libvterm

PhilTaken opened the pull request:
This change allows more fine-grained control over the building process, including overriding dependency versions (see #20406).

PhilTaken committed:
build(nix): change the pkgs to final, add new version of libvterm

This change allows more fine-grained control over the building process,
including overriding dependency versions (see #20406).
It also introduces that overlay to the libvterm-neovim version.

teto commented:
could you prefix the commit with `build(nix):` instead. I would also prefer to see "final"  (or "prev"  when appropriate) replace "pkgs" inplace. I've been using https://github.com/oxalica/nil/releases/tag/2022-09-26 since this week and it also finds a few variables that are unused. If you want to cleanup those at the same time, that would be cool. Also feel free to bump nixpkgs if relevant.

PhilTaken commented:
> could you prefix the commit with `build(nix):` instead. I would also prefer to see "final" (or "prev" when appropriate) replace "pkgs" inplace. I've been using https://github.com/oxalica/nil/releases/tag/2022-09-26 since this week and it also finds a few variables that are unused. If you want to cleanup those at the same time, that would be cool. Also feel free to bump nixpkgs if relevant.

Prefixed the commit, replaced the variables with inplace final, removed an unused binding and ran nixfmt over the whole flake

teto commented on contrib/flake.nix:
```diff
@@ -75,26 +68,22 @@
-              nativeBuildInputs = [ pkgs.shellcheck ];
-              preferLocalBuild = true;
-            } "make -C ${./..} shlint > $out";
+          pylint = pkgs.runCommandNoCC "pylint" {
```
in recent nixpkgs: `runCommandNoCC = throw "'runCommandNoCC' has been renamed to/replaced by 'runCommand'"; # Converted to throw 2022-09-24`  use runCommand instead

PhilTaken replied:
> no rule to make target `shlint`

Changed it to runCommand though

teto replied:
you've done well enough already.

PhilTaken commented:
> Works for me, if you have nothing to add, tell me and I merge.

I thinks that's it. I made sure to overwrite the libvterm-neovim package just for neovim-unwrapped in this overlay to not break other packages outside this scope.

Builds for me and besides shlint all checks pass. I'd say it's ready to be merged.

polarmutex commented:
this version is not building for me using the nixpkgs wrapper. it shows version 7.2 and not master as I would suspect.

teto commented:
the flake contains only an unwrapped version named "master"  so in your nix store it would point at /nix/store/HASH-neovim-master. You've probably misconfigured something. You can import nixpkgs with the flake overlay or if you use home-manager set neovim.package to the unwrapped version in the flake

polarmutex commented:
I was going commit by commit in the neovim repo in my flake. And when I get to this change it went from master to 7.2 version. I will try to track it down further but just reporting what I am noticing
